src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -122,12 +123,21 @@ class HierarchicalNSW : public VecsimBaseObject {
                                    candidatesMaxHeap<dist_t> &top_candidates,
                                    candidatesMaxHeap<dist_t> &candidates_set,
                                    dist_t lowerBound) const;
+    inline void processCandidate_RangeSearch(tableint curNodeId, const void *data_point,
+                                             size_t layer, float epsilon, tag_t visited_tag,
+                                             VecSimQueryResult **top_candidates,
+                                             candidatesMaxHeap<dist_t> &candidate_set,
+                                             dist_t lowerBound, float radius) const;
```

Comment:
shouldn't the radius be of type `dist_t`?

Reply:
It is sent as `float` from the library API:
(`VecSimQueryResult_List VecSimIndex_RangeQuery(
VecSimIndex *index,
 const void *queryBlob,
float radius,
VecSimQueryParams *queryParams,
VecSimQueryResult_Order order)`

since flat supports only float currently. I guess that it we can change `radius` to be generic as part of the supporting `float64` vectors PR.

Reply:
in this file, every parameter related to the distance value should be of type `dist_t`